actions/reservation.py

```diff
@@ -0,0 +1,31 @@
+from __future__ import annotations
+
+import datetime
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from reservation_units.models import ReservationUnit
+    from reservations.models import Reservation
+
+
+class ReservationActions:
+    def __init__(self, reservation: Reservation):
+        self.reservation = reservation
+
+    def get_actual_before_buffer(self) -> datetime.timedelta:
+        buffer_time_before: datetime.timedelta = self.reservation.buffer_time_before or datetime.timedelta()
+        reservation_unit: ReservationUnit
+        for reservation_unit in self.reservation.reservation_unit.all():
+            before = reservation_unit.actions.get_actual_before_buffer(self.reservation.begin)
+            if before > buffer_time_before:
+                buffer_time_before = before
+        return buffer_time_before
+
+    def get_actual_after_buffer(self) -> datetime.timedelta:
+        buffer_time_after: datetime.timedelta = self.reservation.buffer_time_after or datetime.timedelta()
+        reservation_unit: ReservationUnit
+        for reservation_unit in self.reservation.reservation_unit.all():
+            after = reservation_unit.actions.get_actual_after_buffer(self.reservation.end)
+            if after > buffer_time_after:
+                buffer_time_after = after
+        return buffer_time_after
```

Comment:
Add comments why these helpers are needed

Reply:
I don't think these were used in the end. The idea was to calculate the reservation's actual current reservable time, and not the one at the time of creation, but based on previous tests, it seems that the set values should be honored (e.g. there was a case where staff reservations had 0 buffer set despite unit settings). @vergama might want to comment on what should be done here.

Reply:
To clarify:

If something other than the time of an existing reservation is modified: the already set buffer values on it are honored.

If a customer user changes the time of a reservation, the new version should get the buffers set again, according to the current reservation unit settings (whether it is "whole day blocking" or has specific buffer lengths) -- since that corresponds to cancelling the existing reservation and creating a new one, as far as its target time is considered.

If a staff user changes the time of a reservation:
* Reservation unit is (at least now) "whole day blocking": the staff user has no choice and the buffers are set anew accordingly (as the implementation now is).
* Reservation unit has some specific buffer lengths (zero or otherwise, same or not as when the reservation was originally created): staff user has a choice of what buffers lengths to apply to the new version of the reservation (need not be constrained in backend); UI is supposed to guide the choice between 0 or current res.unit level setting ("no buffer" vs. "same as customer user would get") -- the last part doesn't actually quite work so but that is a frontend issue.

Reply:
Yes.
At least for now it is valid that it is so (the existing buffers saved to the reservations are used, even if they no longer correspond to current and corresponding buffer settings at the reservation unit level).